File: lib/widgets/course_table/course_table_widget_state.dart
import 'package:flutter/cupertino.dart';
import 'package:swustmeow/entity/soa/course/course_entry.dart';

class CourseTableWidgetState {
  final success = ValueNotifier(true);
  final lastUpdateTimestamp =
      ValueNotifier(DateTime.now().millisecondsSinceEpoch);
  final weekNum = ValueNotifier(0);
  final entries = ValueNotifier<List<CourseEntry>?>(null);
  final termStartDate = ValueNotifier<DateTime>(DateTime.now());
  final courseTableTimes = ValueNotifier<List<String>>([]);
  final term = ValueNotifier<String>('');

  void clear() {
    entries.value = null;
  }
}

File: android/app/src/main/kotlin/store/swust/swustmeow/widgets/course_table/CourseTableWidgetState.kt
package store.swust.swustmeow.widgets.course_table

import store.swust.swustmeow.entities.SimpleCourseEntry
import java.util.Date

data class CourseTableWidgetState(
    val success: Boolean = false,
    val lastUpdateTimestamp: Long = 0,
    val weekNum: Int = 0,
    val entries: List<SimpleCourseEntry>? = null,
    val termStartDate: Date = Date(),
    val courseTableTimes: List<String> = emptyList()
)